TypeCobol/Compiler/CodeElements/Paragraph/RepositoryParagraph.cs

```diff
@@ -1,21 +1,46 @@
 ﻿using System;
 using System.Collections.Generic;
+using TypeCobol.Compiler.Scanner;
 
 namespace TypeCobol.Compiler.CodeElements
 {
     /// <summary>
     /// The REPOSITORY paragraph is used in a program or class definition to identify all
     /// the object-oriented classes that are intended to be referenced in that program or
-    /// class definition.
+    /// class definition. It also allows declaration of intrinsic function names that may be used without
+    /// specifying the keyword FUNCTION.
     /// </summary>
     public class RepositoryParagraph : CodeElement
     {
+        /// <summary>
+        /// Intrinsic functions which are not allowed in the REPOSITORY paragraph
+        /// </summary>
+        public static readonly HashSet<string> NotAllowedIntrinsicFunctions = new[] { "WHEN-COMPILED" }.ToHashSet(StringComparer.OrdinalIgnoreCase);
```

Comment:
Use collection initializer instead of `ToHashSet` method.

Reply:
This should not be public.

Reply:
Move this to CrossChecker so no need to have public access here

Reply:
Collection initializer used + moved to CrossChecker but in a dedicated class for Repository checking.